projects/ad9361/src/ad9361_api.c

```diff
@@ -125,7 +126,7 @@ int32_t ad9361_init (struct ad9361_rf_phy **ad9361_phy,
 	phy->dev_sel = init_param->dev_sel;
 
 	/* Identification number */
-	phy->spi->id = init_param->id_no;
+	((xil_spi_desc*)phy->spi->extra)->id = init_param->id_no;
```

Comment:
The API must remain platform agnostic.

Reply:
I think for the Xilinx projects a general direction should be established, otherwise we just keep modifying things all around with no consistency.